### ISSUE TITLE
Test suite doesn't work with install script installation

### ISSUE THREAD

**Issue by miklos1** [firedrakeproject/firedrake#624]
I have also tried with `--devel`.

The errors are usually PETSc-related:
```
_______________________ ERROR collecting tests/regression/test_vector_laplace_on_quadrilaterals.py ________________________
tests/regression/test_vector_laplace_on_quadrilaterals.py:4: in <module>
>   from firedrake import *
firedrake/__init__.py:3: in <module>
>   import firedrake.petsc as petsc
firedrake/petsc.py:5: in <module>
>   petsc4py.init(sys.argv)
/homes/mh1714/.local/lib/python2.7/site-packages/petsc4py/__init__.py:42: in init
>       PETSc = petsc4py.lib.ImportPETSc(arch)
/homes/mh1714/.local/lib/python2.7/site-packages/petsc4py/lib/__init__.py:29: in ImportPETSc
>       return Import('petsc4py', 'PETSc', path, arch)
/homes/mh1714/.local/lib/python2.7/site-packages/petsc4py/lib/__init__.py:64: in Import
>       module = imp.load_module(fullname, fo, fn, stuff)
E       ImportError: libpetsc.so.3.05: cannot open shared object file: No such file or directory
```

Otherwise Firedrake seems to work.

**Comment by miklos1** [comment]
```
pip install --upgrade pytest
```
in the virtualenv seems to solve it. Thanks @wence-.

**Comment by wence-** [comment]
Should the install script do this such that a user can run the test suite? (i.e. so that they get the "correct" pytest)?

**Comment by miklos1** [comment]
Possibly. @dham?

**Comment by dham** [comment]
Yes.

**Comment by wence-** [comment]
Fixed in #698.
